on_the_porch/data_ingestion/boston_data_sync/boston_data_sync.py

```
@@ -0,0 +1,1150 @@
+            date_from = None
+            date_to = datetime.now().strftime('%Y-%m-%d')  # Today
+            date_field_normalized = date_field.replace(' ', '_').replace('-', '_').lower() if date_field else None
+            
+            # Check if table exists and has data - if so, do incremental sync
+            cursor.execute(f"SHOW TABLES LIKE '{table_name}'")
+            table_exists = cursor.fetchone() is not None
+            
+            if table_exists and date_field_normalized:
+                try:
+                    # Get the latest date from the database
+                    cursor.execute(f"SELECT MAX(`{date_field_normalized}`) as max_date FROM `{table_name}`")
```

Ruff (0.14.6)

Possible SQL injection vector through string-based query construction

(S608)